File: PhysTasks/Task4.py
import math

g = 9.81

V = float(input("Введите начальную скорость V (м/с): "))
alpha = float(input("Введите угол α (в градусах): "))
H = float(input("Введите высоту H (м): "))

alpha_rad = math.radians(alpha)

T = (V * math.sin(alpha_rad)) / g * (1 + math.sqrt(1 + (2 * g * H) / (V ** 2 * math.sin(alpha_rad) ** 2)))

print(f"Время полета T: {T:.2f} секунд")

File: PhysTasks/Task4.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    const double g = 9.81; 
    
    double V, alpha, H;

    cout << "Введите начальную скорость V (м/с): ";
    cin >> V;
    cout << "Введите угол α (в градусах): ";
    cin >> alpha;
    cout << "Введите высоту H (м): ";
    cin >> H;

    double alpha_rad = alpha * M_PI / 180.0;

    double T = (V * sin(alpha_rad)) / g * (1 + sqrt(1 + (2 * g * H) / (V * V * sin(alpha_rad) * sin(alpha_rad))));

    cout << "Время полета T: " << T << " секунд" << endl;

    return 0;
}
